dbms/src/Functions/greatCircleDistance.cpp

```diff
@@ -38,7 +119,7 @@ class FunctionGreatCircleDistance : public IFunction
 public:
 
     static constexpr auto name = "greatCircleDistance";
-    static FunctionPtr create(const Context &) { return std::make_shared<FunctionGreatCircleDistance>(); }
+    static FunctionPtr create(const Context &) { GeodistInit(); return std::make_shared<FunctionGreatCircleDistance>(); }
```

Comment:
`create()` is called in a query context, so calling `GeodistInit()` here might introduce a ThreadSanitizer warning, if two queries with this functions are executed simultaneously. A good place to do the initialization of global constants would be `registerFunction`, which is called at server startup in a single-threaded context.